cli/azd/cmd/container.go

```diff
@@ -476,6 +517,16 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		}
 	}
 
+	// Function to determine the default IaC provider when provisioning
+	container.RegisterSingleton(func(
+		lazyProjectConfig *lazy.Lazy[*project.ProjectConfig],
+		configManager config.UserConfigManager,
+	) provisioning.DefaultProviderResolver {
+		return func() (provisioning.ProviderKind, error) {
+			return provisioning.Bicep, nil
+		}
+	})
```

Comment:
The `provisioning` package doesn't actually even know anything about the devcenter implementation so it would be the wrong place to make this decision.  The default provider is a higher level application concern and in that case feel like the new platform configuration concept is the correct location to make these application wide configuration changes.